src/Entities/Painting.cpp

```diff
@@ -41,3 +43,11 @@ void cPainting::GetDrops(cItems & a_Items, cEntity * a_Killer)
 
 
 
+
+void cPainting::KilledBy(TakeDamageInfo & a_TDI)
+{
+	Super::KilledBy(a_TDI);
+	Destroy();
+
+	m_World->BroadcastSoundEffect("entity.painting.break", GetPosition(), 1, 1);
```

Comment:
Destroy should be moved to the end of this function, as it frees the memory used by the entity and so calling GetPosition is a use-after-free.